test/instrumentations/grpc-js/grpc-js.test.ts

```diff
@@ -0,0 +1,233 @@
+        expect(clientSpan.attributes["rpc.method"]).toEqual(method);
+
+        expect(clientSpan["traceId"]).toEqual(serverSpan["traceId"]);
+        expect(clientSpan["spanId"]).toEqual(serverSpan["parentSpanId"]);
+
+        return [
+            serverSpan.attributes["rpc.request.payload"],
+            clientSpan.attributes["rpc.request.payload"],
+            serverSpan.attributes["rpc.response.payload"],
+            clientSpan.attributes["rpc.response.payload"]
+        ];
+    }
+
+    itTest(
+      {
+        testName: `roundtrip unary/unary: ${versionToTest}`,
+        packageName: 'grpc-js',
+        version: versionToTest,
+        timeout: TEST_TIMEOUT,
+      },
+      async function () {
+        const exporterFile = `${SPANS_DIR}/server-roundtrip-unary-unary-grpc-js@${versionToTest}.json`;
+
+        testApp = new TestApp(TEST_APP_DIR, INSTRUMENTATION_NAME, exporterFile, {
+          OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT: '4096',
+        });
+
+        const grpcPort = DEFAULT_GRPC_PORT;
+
+        await testApp.invokeGetPath(`/start-server?port=${grpcPort}`);
+
+        const greetingName = 'Siri';
+
+        await testApp.invokeGetPath(
+          `/make-unary-unary-request?port=${grpcPort}&name=${greetingName}`
+        );
+
+        const [serverRequest, clientRequest, serverResponse, clientResponse] = await checkSpans(exporterFile, "SayHelloUnaryUnary");
+        expect(serverRequest).toEqual('{"name":"Siri"}');
+        expect(clientRequest).toEqual('{"name":"Siri"}');
+        expect(serverResponse).toEqual('{"message":"Hello Siri"}');
+        expect(clientResponse).toEqual('{"message":"Hello Siri"}');
+      }
+    );
+
+    itTest(
+      {
+        testName: `roundtrip unary/stream: ${versionToTest}`,
+        packageName: 'grpc-js',
+        version: versionToTest,
+        timeout: TEST_TIMEOUT,
+      },
+      async function () {
+        const exporterFile = `${SPANS_DIR}/server-roundtrip-unary-stream-grpc-js@${versionToTest}.json`;
+
+        testApp = new TestApp(TEST_APP_DIR, INSTRUMENTATION_NAME, exporterFile, {
+          OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT: '4096',
+        });
+
+        const grpcPort = DEFAULT_GRPC_PORT;
+
+        await testApp.invokeGetPath(`/start-server?port=${grpcPort}`);
+
+        const greetingName = 'Siri';
+
+        await testApp.invokeGetPath(
+          `/make-unary-stream-request?port=${grpcPort}&name=${greetingName}`
+        );
+
+        const [serverRequest, clientRequest, serverResponse, clientResponse] = await checkSpans(exporterFile, "SayHelloUnaryStream");
+        expect(serverRequest).toEqual('{"name":"Siri"}');
+        expect(clientRequest).toEqual('{"name":"Siri"}');
+        // TODO: collect the response stream from the server - RD-11068
+        // expect(serverResponse).toEqual('{"message":"Hello Siri 0"}{"message":"Hello Siri 1"}{"message":"Hello Siri 2"}{"message":"Hello Siri 3"}{"message":"Hello Siri 4"}');
```

Comment:
is this both python and node?

Reply:
yes.
checkout the ticket for full description:
https://lumigo.atlassian.net/browse/RD-11068